File: app/src/main/java/com/imangazalievm/bubbble/di/global/ApplicationComponent.java
package com.imangazalievm.bubbble.di.global;

import com.imangazalievm.bubbble.di.modules.OkHttpInterceptorsModule;
import com.imangazalievm.bubbble.domain.global.repository.CommentsRepository;
import com.imangazalievm.bubbble.domain.global.repository.FollowersRepository;
import com.imangazalievm.bubbble.domain.global.repository.ImagesRepository;
import com.imangazalievm.bubbble.domain.global.repository.ShotsRepository;
import com.imangazalievm.bubbble.domain.global.repository.UsersRepository;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {ApplicationModule.class, DataModule.class, OkHttpInterceptorsModule.class})
public interface ApplicationComponent {

    ShotsRepository provideBookRepository();

    CommentsRepository provideCommentsRepository();

    UsersRepository provideUsersRepository();

    FollowersRepository provideFollowersRepository();

    ImagesRepository provideImagesRepository();

}

File: app/src/main/java/com/imangazalievm/bubbble/di/global/ApplicationComponent.kt
package com.imangazalievm.bubbble.di.global

import com.imangazalievm.bubbble.data.shots.CommentsRepository
import com.imangazalievm.bubbble.data.shots.ImagesRepository
import com.imangazalievm.bubbble.data.shots.ShotsRepository
import com.imangazalievm.bubbble.data.users.FollowersRepository
import com.imangazalievm.bubbble.data.users.UsersRepository
import com.imangazalievm.bubbble.di.modules.OkHttpInterceptorsModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(
    modules = [
        ApplicationModule::class,
        DataModule::class,
        OkHttpInterceptorsModule::class
    ]
)
interface ApplicationComponent {
    fun provideBookRepository(): ShotsRepository
    fun provideCommentsRepository(): CommentsRepository
    fun provideUsersRepository(): UsersRepository
    fun provideFollowersRepository(): FollowersRepository
    fun provideImagesRepository(): ImagesRepository
}